### PR TITLE
[MNOE-952] [MNOE-1042] ensure read only for Account Manager role

### PR DESCRIPTION
**Pull request by paulstuartparker**

### CODE REVIEW

**Comment by adamaziz15** [comment]
The only improvement I could think of is making a function in `current-user.svc.coffee` called `@isStaff` to dry up the repeated code (even though it doesn't save much space). @paulstuartparker

File: src/app/views/products/product.controller.coffee [hunk 1]
```diff
@@ -7,6 +7,11 @@
   vm.product.pricing_plans = []
   vm.currencies = _.clone(CURRENCIES.values)
 
+  MnoeCurrentUser.getUser().then(
```
Comment (adamaziz15) [inline src/app/views/products/product.controller.coffee]: see comment about refactoring these ^

File: src/app/views/dashboard.layout.html [hunk 2]
```diff
@@ -76,6 +76,12 @@
     <li class="sidebar-list" ng-if="::main.isDashboardTemplatesEnabled" ui-sref-active="active">
       <a ui-sref="dashboard.dashboard-templates">{{'mnoe_admin_panel.dashboard.layout.sidebar.dashboard_templates' | translate}}<span class="menu-icon fa fa-bar-chart"></span></a>
     </li>
+    <div ng-if="::main.adminConfig.areSettingsEnabled()" ng-show="main.user.admin_role == 'staff'">
```
Comment (ouranos) [inline src/app/views/dashboard.layout.html]: Duplicate code.
Refactor the menu below to only display the "settings-apps" submenu when `role == 'staff'`

File: src/app/views/settings/apps/apps.html [hunk 3]
```diff
@@ -9,12 +9,12 @@
               <div class="row apps-box">
                 <div class="app-wrapper col-xs-3 col-sm-2 col-md-2" ng-repeat="app in vm.enabledApps">
                   <div class="small-app-card ellipsis">
-                    <i class="fa fa-trash app-remove" ng-if="vm.tenantManagement" ng-click="vm.openRemoveAppModal(app, $index)" uib-tooltip="{{'mnoe_admin_panel.dashboard.organization.remove_app' | translate:{'app_name': app.name} }}" tooltip-append-to-body="true"></i>
+                    <i class="fa fa-trash app-remove" ng-if="vm.tenantManagement" ng-hide="vm.isAccountManager" ng-click="vm.openRemoveAppModal(app, $index)" uib-tooltip="{{'mnoe_admin_panel.dashboard.organization.remove_app' | translate:{'app_name': app.name} }}" tooltip-append-to-body="true"></i>
```
Comment (ouranos) [inline src/app/views/settings/apps/apps.html]: Could we bundle the conditions together? I don't like mixing `ng-if` and `ng-hide`.
Might have to define a method in the controller to keep it cleaner

File: src/app/views/settings/apps/apps.html [hunk 4]
```diff
@@ -9,12 +9,12 @@
...
                 <div class="app-wrapper col-xs-3 col-sm-2 col-md-2" ng-repeat="app in vm.enabledApps">
                   <div class="small-app-card ellipsis">
-                    <i class="fa fa-trash app-remove" ng-if="vm.tenantManagement" ng-click="vm.openRemoveAppModal(app, $index)" uib-tooltip="{{'mnoe_admin_panel.dashboard.organization.remove_app' | translate:{'app_name': app.name} }}" tooltip-append-to-body="true"></i>
+                    <i class="fa fa-trash app-remove" ng-if="vm.tenantManagement && !vm.isAccountManager" ng-click="vm.openRemoveAppModal(app, $index)" uib-tooltip="{{'mnoe_admin_panel.dashboard.organization.remove_app' | translate:{'app_name': app.name} }}" tooltip-append-to-body="true"></i>
                     <img ng-src="{{::app.logo}}" width="60">
                     <span>{{::app.name}}</span>
                   </div>
                 </div>
-                <div ng-if="vm.tenantManagement" class="app-wrapper col-xs-3 col-sm-2 col-md-2">
+                <div ng-if="vm.tenantManagement" ng-hide="vm.isAccountManager" class="app-wrapper col-xs-3 col-sm-2 col-md-2">
```
Comment (adamaziz15) [inline src/app/views/settings/apps/apps.html]: Can we combine these conditions as well? `ng-if="vm.tenantManagement && !vm.isAccountManager"` just like line 12 ^

File: src/app/views/dashboard.layout.html [hunk 5]
```diff
@@ -76,6 +76,12 @@
     <li class="sidebar-list" ng-if="::main.isDashboardTemplatesEnabled" ui-sref-active="active">
       <a ui-sref="dashboard.dashboard-templates">{{'mnoe_admin_panel.dashboard.layout.sidebar.dashboard_templates' | translate}}<span class="menu-icon fa fa-bar-chart"></span></a>
     </li>
+    <div ng-if="main.user.admin_role == 'staff'">
```
Comment (adamaziz15) [inline src/app/views/dashboard.layout.html]: Was this refactored? I think the idea was to have just one settings menu. Would something like this work?: 
```
<div ng-if="::main.adminConfig.areSettingsEnabled()" ng-show="['admin', 'staff'].includes(main.user.admin_role)">
      <li class="sidebar-title"><span>{{'mnoe_admin_panel.dashboard.layout.sidebar.settings' | translate}}</span></li>
      <li class="sidebar-list" ui-sref-active="active" ng-hide="main.user.admin_role == 'staff'">
        <a ui-sref="dashboard.settings-general">{{'mnoe_admin_panel.dashboard.layout.sidebar.settings-general' | translate}}<span class="menu-icon fa fa-cogs"></span></a>
      </li>
      <li class="sidebar-list" ui-sref-active="active">
        <a ui-sref="dashboard.settings-apps">{{'mnoe_admin_panel.dashboard.layout.sidebar.settings-apps' | translate}}<span class="menu-icon fa fa-th-large"></span></a>
      </li>
      <li class="sidebar-list" ui-sref-active="active" ng-hide="main.user.admin_role == 'staff'">
        <a ui-sref="dashboard.settings-plugins">{{'mnoe_admin_panel.dashboard.layout.sidebar.settings-plugins' | translate}}<span class="menu-icon fa fa-puzzle-piece"></span></a>
      </li>
      <li class="sidebar-list" ui-sref-active="active" ng-hide="main.user.admin_role == 'staff'">
        <a ui-sref="dashboard.settings-domain">{{'mnoe_admin_panel.dashboard.layout.sidebar.settings-domain' | translate}}<span class="menu-icon fa fa-shield"></span></a>
      </li>
</div>
```

**Comment by ouranos** [comment]
This has been merged in #268

**Pull request closed without merging**